src/textual/css/stylesheet.py

```diff
@@ -437,6 +439,23 @@ def apply(
         rule_attributes: defaultdict[str, list[tuple[Specificity6, object]]]
         rule_attributes = defaultdict(list)
 
+        if cache is not None:
+            cache_key: tuple | None = (
+                node._parent,
+                node._id,
+                node.classes,
+                node.pseudo_classes,
+                node._css_type_name,
+            )
+            assert cache_key is not None
```

Comment:
If you “declare” `cache_key: tuple | None` before the `if`, can't you get rid of this `assert`?